### PR TITLE
Feature/oas config env generator

### PR DESCRIPTION
**Pull request by danistrebel**
## Description
What's changed, or what was fixed?

- OAS based configurable proxy generator

## Housekeeping
(please check all that apply [X], do not edit the text)
- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.
- [ ] PR requires full pipeline run (Run for changes only by default).

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

File: tools/oas-configurable-proxy/index.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        const api = await SwaggerParser.validate(config.oasPath);
+        api.openapi
+        const oasVersionString = api.openapi || api.swagger;
+        const oasVersionSplit = oasVersionString.split(".");
+        const oasVersionMajor = parseInt(oasVersionSplit[0], 10);
+        const proxyName = config.name || api.info.title.replace(/ /g, '-').toLowerCase();
+        const basepath = config.basepath || `/${proxyName}/v1`
+
+        // target service
+        let targetService = 'https://notfound.apigee.google.com # TODO replace me'
```
Comment (OmidTahouri) [inline tools/oas-configurable-proxy/index.js:49]: Is the # intentionally inside the string or is that supposed to be a JS line comment? :)

File: tools/oas-configurable-proxy/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "oas-configurable-proxy",
+  "version": "1.0.0",
+  "description": "Generates Apigee Configurable Proxy Envs from Swagger Specs",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (OmidTahouri) [inline tools/oas-configurable-proxy/package.json:7]: You could just execute pipeline.sh as the test, but meh

**Review by OmidTahouri — CHANGES_REQUESTED** [review]

File: tools/oas-configurable-proxy/package.json (hunk 2 above)
Comment (OmidTahouri) [reply tools/oas-configurable-proxy/package.json:7]: Ignore this. No added value in it.

File: tools/oas-configurable-proxy/index.js (hunk 1 above)
Comment (danistrebel) [reply tools/oas-configurable-proxy/index.js:49]: it's a comment to go into the templated YAML ;)

**Review by OmidTahouri — APPROVED** [review]

**Pull request merged**